framework/source/class/qx/util/DynamicScriptLoader.js

```diff
@@ -0,0 +1,231 @@
+
+    /**
+     * fired when all given scripts are loaded, each time loadScriptsDynamic is called
+     */
+    ready: 'qx.event.type.Event'
+  },
+
+  statics: {
+    __ADDED: {},
+    __LOADED: {}
+  },
+
+  /*
+  *****************************************************************************
+     MEMBERS
+  *****************************************************************************
+  */
+
+
+  members: {
+
+    __QUEUE: null,
+    __LOADING: null,
+
+
+    /**
+     * Load scripts dynamically, typically used in the instance constructor
+     *
+     * @param codeArr {Array} an array with the uri names of the scripts
+     * @param callback {Function?undefined} a callback function to call when all the scripts in the array have been loaded
+     * @param context {Object?undefined} a contex for calling the callback
+     *
+     * @return {String|null} name of the last script to be loaded
+     *     Returns the name of the last script you can expect to be loaded if all scripts are already loaded null will be returned.
+     *
+     */
+    load: function(codeArr) {
+      codeArr.forEach(function(script) {
+         this.__QUEUE.push(script);
+      },this);
+      this.__loadScripts();
+    },
+
+
+    /**
+     * Chain loading scripts.
+     *
+     * Recursively called until the array of scripts is consumed
+     *
+     * @param codeArr {Array} an array with the uri names of the scripts
+     */
+    __loadScripts: function() {
+      var cl = qx.util.DynamicScriptLoader;
+      if (this.__LOADING === true) {
+        return;
+      }
+
+      var script = this.__QUEUE.shift();
+      if (!script){
+        this.fireEvent("ready")
+        return;
+      }
+
+      if (cl.__LOADED[script]){
+        this.__loadScripts();
+        return;
+      }           
+
+      this.__LOADING = true
+
+      var dynLoader = cl.__ADDED[script];
+      if (dynLoader){
+          var id1 = dynLoader.addListener('loaded',function(e){
+            var data = e.getData();
+            if (data.script === script){
+              dynLoader.removeListenerById(id2);
+              dynLoader.removeListenerById(id1);
+              this.fireDataEvent('loaded',data);
+              this.__LOADING = false;
+              this.__loadScripts();
+            }
+          },this);
+          var id2 = dynLoader.addListener('failed',function(e){
+            var data = e.getData();
+            if (data.script === script){
+              dynLoader.removeListenerById(id1);
+              dynLoader.removeListenerById(id2);              
+              this.fireDataEvent('failed',data);
+              this.__LOADING = false;
+            }
+          },this);
+          return;
+      }
+
+      var uri = qx.util.ResourceManager.getInstance().toUri(script);
+      var loader = new qx.bom.request.Script();
+
+      loader.on("load", function(request) {
+        cl.__LOADED[script] = true;
+        delete cl.__ADDED[script];
+        this.fireDataEvent('loaded', {
```

Comment:
on line 176 you get a loaded event for instances that piggyback on the loading initiated by this instance. here is the original one ... triggering the one in 176